### ISSUE TITLE
Some tests are failing under Java 21

### ISSUE THREAD

**Issue by ryanprayogo** [open-feature/java-sdk#1059]
Some tests that are using Mockito are failing under Java 21 with this error
```
Cannot call abstract real method on java object!
Calling real methods is only possible when mocking non abstract method.
```
This seem to be caused by an [open issue in Mockito](https://github.com/mockito/mockito/issues/3121).

There is a workaround available (set the `net.bytebuddy.experimental` system property to `true`), which can be set in the unit tests' `@BeforeAll` annotated method.

Steps to reproduce:
- Use Java 21 (reproducible with Temurin JDK 21.0.4)
- Run `mvn clean test`
- Observe failures

**Comment by Mehamoodshaik** [comment]
Try to add this plugin. It might works

<plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-surefire-plugin</artifactId>
            <version>3.0.0-M7</version> 
            <configuration>
                <systemPropertyVariables>
                    <net.bytebuddy.experimental>true</net.bytebuddy.experimental>
                </systemPropertyVariables>
            </configuration>
 </plugin>

**Comment by toddbaert** [comment]
Oh man... this will save me a lot of time! TBH I never tried to troubleshoot this, I just temporarily switched to Java 8.

**Comment by ryanprayogo** [comment]
I have the fix in #1060 

I originally added the said system property in [this commit](https://github.com/open-feature/java-sdk/pull/1060/commits/a75e7056570e48a5de69a605226e8553af4161fa) but end up using the more recent (minor) version of `byte-buddy(-agent)` instead (see discussion in the PR).
